api/commandValidator.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const Joi = require('joi');
+
+// essentially a Joi.ValidationError.details object
+const COMMAND_ERROR = Joi.object()
+    .keys({
+        context: Joi.object()
+            .label('Functional context regarding the error'),
+        message: Joi.string()
+            .label('Description of a particular validation error'),
+        path: Joi.string()
+            .label('Dot-notation path to the field that caused the validation error'),
+        type: Joi.string()
+            .label('The the Joi-type that categorizes the error')
+    })
+    .requiredKeys('message', 'path', 'type');
+
+const SCHEMA_OUTPUT = Joi.object()
+    .keys({
+        errors: Joi.array()
+            .items(COMMAND_ERROR)
+            .label('Array of errors encountered while validating the given command'),
+        // since a command could be parseable but invalid, the contents are unpredicatble
+        command: Joi.object()
```

Comment:
This should be `config` if we will implement `commandValidate` api like `templateValidate`.
https://github.com/screwdriver-cd/command-validator/blob/master/index.js#L54
When I implemented command-validator, I avoided using command because of key duplication.

We should fix response of command-validator to `command` from `config` if it's better.

Reply:
I guess your suggestion is better ...
But, if we use `config` instead of `command`, we will code `config.config` in `commands` plugin.
Ref.: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/templates/create.js#L39

We are in a dilemma.

Reply:
Yea, I think `command` is a little confusing since we have nested commands under it:
```
command:
    namespace: foo
    command: bar
    ...
    habitat:
        mode: remote
        package: core/git/2.14.1
        command: git
```

I feel like inside https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/templates/create.js#L27 you can just name that returned config something else, like `validConfig`. It's a variable in that block of code.

Reply:
`command` key rename PR has been merged.
https://github.com/screwdriver-cd/screwdriver/pull/829
This confusion will be resolved.
```
*command*:
    namespace: foo
    *name*: bar
    ...
    habitat:
        *command*: git
```
So, we can use `command` object instead of the existing `config` in the screwdriver-command-validator and use `config.command` as is in this API commandValidator schema.

Reply:
Ok got it 👍